llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -2679,7 +2679,11 @@ Error BitcodeReader::parseTypeTableBody() {
           return error("Integer parameter too large");
         IntParams.push_back(Record[i]);
       }
-      ResultTy = TargetExtType::get(Context, TypeName, TypeParams, IntParams);
+      auto TTy =
+          TargetExtType::getOrError(Context, TypeName, TypeParams, IntParams);
+      if (auto E = TTy.takeError())
+        return error(toString(std::move(E)));
```

Comment:
This is a bit of a faff - it unwraps the `Error` and then wraps it again, just to change the error code to `CorruptedBitcode`.

Reply:
I think it would be okay to just directly return the error. Nobody uses error codes, only the message matters...